### PR TITLE
Branch with all storage work

### PR DESCRIPTION
**Pull request by Sairsey**

### CODE REVIEW

File: src/MangaJet/app/src/main/java/com/mangajet/mangajet/mangareader/MangaReaderViewModel.kt [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
 import com.mangajet.mangajet.data.Manga
 
 // Class which represents "Manga Reader" ViewModel
-class MangaReaderViewModel : ViewModel {
-    val manga: Manga
+class MangaReaderViewModel : ViewModel() {
+    // ViewModel initialization flag
+    var isInited = false
+    // Manga we are reading right now
+    lateinit var manga: Manga
```
Comment (AS2) [inline src/MangaJet/app/src/main/java/com/mangajet/mangajet/mangareader/MangaReaderViewModel.kt:13]: comment

Comment (AS2) [reply src/MangaJet/app/src/main/java/com/mangajet/mangajet/mangareader/MangaReaderViewModel.kt:13]: private

File: src/MangaJet/app/src/main/java/com/mangajet/mangajet/ui/history/HistoryFragment.kt [hunk 2]
```diff
@@ -39,17 +58,19 @@ class HistoryFragment : Fragment() {
...
                 val intent = Intent(it, AboutMangaActivity::class.java)
-                intent.putExtra("Manga", historyViewModel.mangas[id.toInt()].toJSON())
+                MangaJetApp.currentManga = historyViewModel.mangas[id.toInt()]
                 startActivity(intent)}
         }
 
         return root
     }
-
+
```
Comment (AS2) [inline src/MangaJet/app/src/main/java/com/mangajet/mangajet/ui/history/HistoryFragment.kt:73]: spaces

File: src/MangaJet/app/src/main/java/com/mangajet/mangajet/ui/history/HistoryViewModel.kt [hunk 3]
```diff
@@ -21,33 +22,58 @@ class HistoryViewModel : ViewModel() {
...
+        }
+        for (path in mangasPaths) {
+            job?.ensureActive()
+            try {
+                mangas.add(
+                    Manga(StorageManager.loadString(path, StorageManager.FileType.MangaInfo))
+                )
+            }
+            catch (ex: MangaJetException) {
+                // nothing too tragic, we just haven`t permission to read or file invalid
```
Comment (AS2) [inline src/MangaJet/app/src/main/java/com/mangajet/mangajet/ui/history/HistoryViewModel.kt:41]: continue

**Review by AS2 — CHANGES_REQUESTED** [review]

**Review by AS2 — APPROVED** [review]

**Review by PopovIV — APPROVED** [review]

**Pull request merged**